src/api.js

```diff
@@ -17,33 +17,433 @@ const getMessages = args => {
   const url = '/messages/states';
   return axios
     .post(url, args)
-    .then(res => ({ count: res.data.count, data: res.data.data.map(parseMessage) }));
+    .then(res => Object.assign(parseMessages(res.data.data), { count: res.data.count }));
```

Comment:
```suggestion
    .then(res => ({...parseMessages(res.data.data), count: res.data.count }));
```

Reply:
count should probably be returned by parseMessages

Reply:
This syntax is not available in es6